zarr/tests/test_convenience.py

```diff
@@ -229,64 +236,81 @@ def test_consolidate_metadata(with_chunk_store, zarr_version):
     arr[:] = 1.0
     assert 16 == arr.nchunks_initialized
 
+    if stores_from_path:
+        # get the actual store class for use with consolidate_metadata
+        store_class = z._store
+    else:
+        store_class = store
+
     if zarr_version == 3:
         # error on v3 if path not provided
         with pytest.raises(ValueError):
-            consolidate_metadata(store, path=None)
+            consolidate_metadata(store_class, path=None)
 
         with pytest.raises(ValueError):
-            consolidate_metadata(store, path='')
+            consolidate_metadata(store_class, path='')
 
     # perform consolidation
-    out = consolidate_metadata(store, path=path)
+    out = consolidate_metadata(store_class, path=path)
     assert isinstance(out, Group)
     assert ['g1', 'g2'] == list(out)
-    if zarr_version == 2:
-        assert isinstance(out._store, ConsolidatedMetadataStore)
-        assert '.zmetadata' in store
-        meta_keys = ['.zgroup',
-                     'g1/.zgroup',
-                     'g2/.zgroup',
-                     'g2/.zattrs',
-                     'g2/arr/.zarray',
-                     'g2/arr/.zattrs']
-    else:
-        assert isinstance(out._store, ConsolidatedMetadataStoreV3)
-        assert 'meta/root/consolidated/.zmetadata' in store
-        meta_keys = ['zarr.json',
-                     meta_root + 'dataset.group.json',
-                     meta_root + 'dataset/g1.group.json',
-                     meta_root + 'dataset/g2.group.json',
-                     meta_root + 'dataset/g2/arr.array.json',
-                     'meta/root/consolidated.group.json']
-    for key in meta_keys:
-        del store[key]
+    if not stores_from_path:
+        if zarr_version == 2:
+            assert isinstance(out._store, ConsolidatedMetadataStore)
+            assert '.zmetadata' in store
+            meta_keys = ['.zgroup',
+                         'g1/.zgroup',
+                         'g2/.zgroup',
+                         'g2/.zattrs',
+                         'g2/arr/.zarray',
+                         'g2/arr/.zattrs']
+        else:
+            assert isinstance(out._store, ConsolidatedMetadataStoreV3)
+            assert 'meta/root/consolidated/.zmetadata' in store
```

Comment:
Again, outwith this PR but back to https://github.com/zarr-developers/zarr-specs/issues/113, could the v3 spec here simply replace all the individual JSON files _solely_ with the consolidated metadata?

Reply:
I think it could be done, but haven't tried it. All store types are already somewhat consolidated for v3 since the entire `meta` folder tree is independent of the `data` one. I suppose if there are many arrays/groups it may still be preferred to further consolidate into one file, though? I would defer to those using the feature on what is the desired behavior.